geode-core/src/test/java/com/gemstone/gemfire/cache/query/QueryWithBucketParameterIntgTest.java

```diff
@@ -0,0 +1,224 @@
+        import com.gemstone.gemfire.internal.cache.LocalDataSet;
+        import com.gemstone.gemfire.internal.cache.PartitionedRegion;
+        import com.gemstone.gemfire.test.junit.categories.IntegrationTest;
+        import org.junit.rules.ExpectedException;
+
+/**
+ * @implNote This test class has been designed to test the query execution using the LocalDataSet.executeQuery
+ * where one of the parameters passed is the bucket set to be used.
+ * Different variation of hashset variables are passed to the function to check for errors.
+ * Created by nnag on 3/3/16.
+ */
+@Category(IntegrationTest.class)
+public class QueryWithBucketParameterIntgTest {
+
+    //Necessary data elements required for execution of the test
+//    Region region;
+//    QueryService qs;
+//    Cache cache;
+    DefaultQuery queryExecutor;
+    LocalDataSet lds;
+
+    /**
+     * @implNote This code is copied from BugJUnitTest.java. These elements are required to pass as parameters
+     * to create data elements for the region
+     * @implNote This needs to be moved to a common test package to avoid redundant code.
+     */
+    static class MyValue implements Serializable, Comparable<MyValue>
+    {
+        public int value = 0;
+
+        public MyValue(int value) {
+            this.value = value;
+        }
+
+
+        public int compareTo(MyValue o)
+        {
+            if(this.value > o.value) {
+                return 1;
+            }else if(this.value < o.value) {
+                return -1;
+            }else {
+                return 0;
+            }
+        }
+    }
+
+    //Empty constructor
+    public QueryWithBucketParameterIntgTest() {
+    }
+
+    /**
+     * Pre-test execution to create the test bed.
+     * This code snippet has been borrowed from (com.gemstone.gemfire.cache.query) BugJUintTest.java [03/02/2016]
+     * @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception {
+        String regionName = "pr1";
+        int totalBuckets = 40;
+        int numValues = 80;
+        CacheUtils.startCache();
+        Cache cache = CacheUtils.getCache();
+        PartitionAttributesFactory pAFactory = getPartitionAttributesFactoryWithPartitionResolver(totalBuckets);
+        RegionFactory rf = cache.createRegionFactory(RegionShortcut.PARTITION);
+        rf.setPartitionAttributes(pAFactory.create());
+        PartitionedRegion pr1 = (PartitionedRegion) rf.create(regionName);
+        populateRegion(pr1, numValues);
+        QueryService qs = pr1.getCache().getQueryService();
+        String query = "select distinct e1.value from /pr1 e1";
+        queryExecutor = (DefaultQuery)CacheUtils.getQueryService().newQuery(
+                query);
+        Set<Integer> set = getBucketList(totalBuckets);
+        lds = new LocalDataSet(pr1, set);
+    }
+
+    private Set<Integer> getBucketList(int nBuckets) {
+        Set<Integer> set = new HashSet<Integer>();
+        for (int i = 0; i < nBuckets; ++i) {
```

Comment:
It might be possible to use IntStream and condense this for loop into  a foreach call instead.

Something like 
IntStream.range(0, nBuckets).forEach(i -> set.add(i));